mypy/checkexpr.py

```diff
@@ -1806,123 +1808,278 @@ def get_operator_method(self, op: str) -> str:
+            ]
+        else:
+            # In all other cases, we do the usual thing and call __add__ first and
+            # __radd__ second when doing "A() + B()".
+
+            variants_raw = [
+                (left_op, left_type, right_expr),
+                (right_op, right_type, left_expr),
+            ]
+
+        # STEP 2b:
+        # When running Python 2, we might also try calling the __cmp__ method.
+
+        is_python_2 = self.chk.options.python_version[0] == 2
+        if is_python_2 and op_name in nodes.ops_falling_back_to_cmp:
+            cmp_method = nodes.comparison_fallback_method
+            left_cmp_op = lookup_operator(cmp_method, left_type)
+            right_cmp_op = lookup_operator(cmp_method, right_type)
+
+            if bias_right:
+                variants_raw.append((right_cmp_op, right_type, left_expr))
+                variants_raw.append((left_cmp_op, left_type, right_expr))
+            else:
+                variants_raw.append((left_cmp_op, left_type, right_expr))
+                variants_raw.append((right_cmp_op, right_type, left_expr))
+
+        # STEP 3:
+        # We now filter out all non-existant operators. The 'variants' list contains
+        # all operator methods that are actually present, in the order that Python
+        # attempts to invoke them.
+
+        variants = [(op, obj, arg) for (op, obj, arg) in variants_raw if op is not None]
+
+        # STEP 4:
+        # We now try invoking each one. If an operation succeeds, end early and return
+        # the corresponding result. Otherwise, return the result and errors associated
+        # with the first entry.
+
+        errors = []
+        results = []
+        for method, obj, arg in variants:
+            local_errors = make_local_errors()
+            result = self.check_op_local(op_name, method, obj, arg, context, local_errors)
+            if local_errors.is_errors():
+                errors.append(local_errors)
+                results.append(result)
+            else:
+                return result
+
+        # STEP 4b:
+        # Sometimes, the variants list is empty. In that case, we fall-back to attempting to
+        # call the __op__ method (even though it's missing).
+
+        if not variants:
+            local_errors = make_local_errors()
+            result = self.check_op_local_by_name(
+                op_name, left_type, right_expr, context, local_errors)
+
+            if local_errors.is_errors():
+                errors.append(local_errors)
+                results.append(result)
+            else:
+                # In theory, we should never enter this case, but it seems
+                # we sometimes do, when dealing with Type[...]? E.g. see
+                # check-classes.testTypeTypeComparisonWorks.
+                #
+                # This is probably related to the TODO in lookup_operator(...)
+                # up above.
+                #
+                # TODO: Remove this extra case
```

Comment:
I think this is something deserving a follow-up issue. Could you please open one?